client/zeth/cli/zeth_mix.py

```diff
@@ -75,18 +77,27 @@ def mix(
     tx_value: Optional[EtherValue] = None
     if mixer_desc.token:
         tx_value = EtherValue(0)
+    else:
+        tx_value = tx_value or vin_pub
```

Comment:
What is the purpose of this line?

Reply:
The purpose is to set the ether value of the transaction based on nthe vout value.
This block is now a bit awkward after a refactor.  I've restructured it.

Reply:
>  I've restructured it.

Thanks for that, that's clearer indeed